cloud/terraform/azure_config_builder_v2.py

```diff
@@ -0,0 +1,332 @@
+                'azurerm_subnet.{}'.format(instance['azurerm_subnet']),
+                'azurerm_public_ip.{}'.format(instance['azurerm_public_ip']),
+            ]
+        }
+
+        self.resources_tf['resource']['azurerm_network_interface'][name] = new_nic
+
+    def __new_azure_vm(self, instance):
+        if 'instance_type' not in instance or not instance['instance_type']:
+            if instance['arch'] == 'x86_64':
+                instance['instance_type'] = self.default_x86_vm_size
+            elif instance['arch'] == 'arm64':
+                instance['instance_type'] = self.default_arm64_vm_size
+
+        instance_hostname = instance['hostname']
+
+        if 'username' in instance:
+            instance_user = instance['username']
+        else:
+            instance_user = self.default_admin_username
+
+        os_disk = {
+            'caching': 'ReadWrite',
+            'storage_account_type': 'Standard_LRS',
+        }
+
+        admin_ssh_key = {
+            'username': instance_user,
+            'public_key': f'${{file("{self.ssh_key_path}")}}'
+        }
+
+        new_instance = {
+            'name': instance_hostname,
+            'location': instance['location'],
+            'admin_username': instance_user,
+            'size': instance['instance_type'],
+            'resource_group_name': self.resource_group,
+            'network_interface_ids': [self.__get_azure_network_resource_uri(
+                terraform_resource_type='azurerm_network_interface',
+                azure_resource_name=instance['azurerm_network_interface'])],
+            'os_disk': os_disk,
+            'admin_ssh_key': admin_ssh_key,
+            'tags': {self.ci_tag_key: self.ci_test_value},
+            'depends_on': [
+                'azurerm_virtual_network.{}'.format(instance['azurerm_virtual_network']),
+                'azurerm_subnet.{}'.format(instance['azurerm_subnet']),
+                'azurerm_network_interface.{}'.format(instance['azurerm_network_interface']),
+            ]
+        }
+
+        if 'image_uri' in instance:
+            new_instance['source_image_id'] = instance['image_uri']
+        elif 'image_definition' in instance:
+            new_instance['source_image_reference'] = instance['image_definition']
+            if 'plan' in instance:
+                new_instance['plan'] = instance['plan']
+        elif 'vhd_uri' in instance:
+            new_instance['depends_on'].append('azurerm_shared_image_version.{}'.format(instance['azurerm_shared_image_version']))
+            new_instance['source_image_id'] = self.__get_azure_image_uri(instance['azurerm_shared_image'])
+
+        self.resources_tf['resource']['azurerm_linux_virtual_machine'][instance_hostname] = new_instance
+
+    def __get_azure_image_uri(self, azure_image_name):
+        """
+        Returns a composed string URI that belongs to a specific Azure image, from its name
```

Comment:
I would specify that this related to a shared image in an Image Gallery.

Reply:
:+1: